src/ModelManager.ts

```diff
@@ -137,7 +136,16 @@ export class ModelManager {
         if (!this._modelClient) {
             this._modelClient = new ModelClient();
         }
-
+        let domain = "";
+        if (this._modelClient) {
+            domain = this._modelClient.apiHost != null ? this._modelClient.apiHost : "";
+        }
+        // Check if async is required. This will be true only in remote rendering.
+        const asyncFlag =  domain === PathUtils.getCurrentPathname() ? false : true;
+        if (asyncFlag && PathUtils.isEditMode()) {
+            const clientLibUrl = this.generateClientLibsUrl(rootModelPath, domain);
+            this.appendClientLibs(clientLibUrl);
+        }
```

Comment:
Is it possible to extract that code in one or mulitpe private functions, for readability?

The following line could be function, etc

`const asyncFlag =  domain === PathUtils.getCurrentPathname() ? false : true;`

Reply:
oks

Reply:
Done.